x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_edit.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RuleAlertType } from './types';
+
+import {
+  BulkActionEditPayload,
+  BulkActionEditType,
+} from '../../../../common/detection_engine/schemas/common/schemas';
+
+import { invariant } from '../../../../common/utils/invariant';
+import { isEqlParams, isQueryParams, isThresholdParams, isThreatParams } from '../signals/utils';
+
+export const addItemsToArray = <T>(arr: T[], items: T[]): T[] =>
+  Array.from(new Set([...arr, ...items]));
+
+export const deleteItemsFromArray = <T>(arr: T[], items: T[]): T[] => {
+  const itemsSet = new Set(items);
+  return arr.filter((item) => !itemsSet.has(item));
+};
+
+// eslint-disable-next-line complexity
+export const applyBulkActionEditToRule = (
+  existingRule: RuleAlertType,
+  action: BulkActionEditPayload
+): RuleAlertType => {
+  const rule = { ...existingRule, params: { ...existingRule.params } };
+
+  switch (action.type) {
+    // tags actions
+    case BulkActionEditType.add_tags:
+      rule.tags = addItemsToArray(rule.tags ?? [], action.value);
+      break;
+
+    case BulkActionEditType.delete_tags:
+      rule.tags = deleteItemsFromArray(rule.tags ?? [], action.value);
+      break;
+
+    case BulkActionEditType.set_tags:
+      rule.tags = action.value;
+      break;
+
+    // index_patterns actions
+    // index pattern is not present in machine learning rule type, so we throw error on it
+    case BulkActionEditType.add_index_patterns:
+      invariant(
+        rule.params.type !== 'machine_learning',
+        "Index patterns can't be added. Machine learning rule doesn't have index patterns property"
+      );
+
+      if (
+        (isEqlParams(rule.params) ||
+          isQueryParams(rule.params) ||
+          isThresholdParams(rule.params) ||
```

Comment:
I think you can use `isMachineLearningParams` here?

Reply:
fixed here: 56bed04c108fc562752a4d34785fbd40702af678